### PR TITLE
feat: split CLI into separate crate

### PR DESCRIPTION
**Pull request by mistydemeo**

### CODE REVIEW

**Comment by Gankra** [comment]
So i think it's an interesting question if we want to tell people to `cargo install axoupdater` or `cargo install axoupdater-cli`... although i suppose admittedly this is a weird crate to cargo-install.

Still part of me feels like the cli is the "main" crate and the lib should have a weird name..?

File: axoupdater-cli/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "axoupdater-cli"
+version = "0.1.0"
+edition = "2021"
+license = "MIT OR Apache-2.0"
+repository = "https://github.com/axodotdev/axoupdater"
```
Comment (Gankra) [inline axoupdater-cli/Cargo.toml:6]: nb both of these Cargo.tomls are missing several fields required for publish

**Comment by mistydemeo** [comment]
Yeah, I was wondering the same. That said: we don't want to tell people to `cargo install` it at all; cargo-dist handles that for them. Instead, people only interact with this from crates.io if they're using it as a library - in which case the `axoupdater` name makes more sense?

**Comment by Gankra** [comment]
Also wait is this not actually creating a cargo workspace here?

**Comment by mistydemeo** [comment]
Yeah, I realized I hadn't actually done that, I'd just split it up into a separate crate. Will restructure into a workspace.

**Comment by Gankra** [comment]
I only noticed because I realized this was a chance for you to mess around with cargo.toml inheritance:

https://doc.rust-lang.org/cargo/reference/workspaces.html#the-package-table

https://github.com/clap-rs/clap/blob/master/clap_derive/Cargo.toml#L14-L17

**Pull request closed without merging**